presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/rest/RestBasedFunctionNamespaceManager.java

```diff
@@ -0,0 +1,243 @@
+    }
+
+    @Override
+    protected Collection<SqlInvokedFunction> fetchFunctionsDirect(QualifiedObjectName functionName)
+    {
+        return latestFunctions.values().stream()
+                .filter(function -> function.getSignature().getName().equals(functionName))
+                .map(RestBasedFunctionNamespaceManager::copyFunction)
+                .collect(toImmutableList());
+    }
+
+    @Override
+    protected UserDefinedType fetchUserDefinedTypeDirect(QualifiedObjectName typeName)
+    {
+        return userDefinedTypes.get(typeName);
+    }
+
+    @Override
+    protected FunctionMetadata fetchFunctionMetadataDirect(SqlFunctionHandle functionHandle)
+    {
+        return fetchFunctionsDirect(functionHandle.getFunctionId().getFunctionName()).stream()
+                .filter(function -> function.getRequiredFunctionHandle().equals(functionHandle))
+                .map(this::sqlInvokedFunctionToMetadata)
+                .collect(onlyElement());
+    }
+
+    @Override
+    protected ScalarFunctionImplementation fetchFunctionImplementationDirect(SqlFunctionHandle functionHandle)
+    {
+        return fetchFunctionsDirect(functionHandle.getFunctionId().getFunctionName()).stream()
+                .filter(function -> function.getRequiredFunctionHandle().equals(functionHandle))
+                .map(this::sqlInvokedFunctionToImplementation)
+                .collect(onlyElement());
+    }
+
+    @Override
+    public void createFunction(SqlInvokedFunction function, boolean replace)
+    {
+        checkFunctionLanguageSupported(function);
+        SqlFunctionId functionId = function.getFunctionId();
+        if (!replace && latestFunctions.containsKey(function.getFunctionId())) {
+            throw new PrestoException(GENERIC_USER_ERROR, format("Function '%s' already exists", functionId.getId()));
+        }
+
+        SqlInvokedFunction replacedFunction = latestFunctions.get(functionId);
+        long version = 1;
+        if (replacedFunction != null) {
+            version = parseLong(replacedFunction.getRequiredVersion()) + 1;
+        }
+        latestFunctions.put(functionId, function.withVersion(String.valueOf(version)));
+    }
+
+    @Override
+    public void alterFunction(QualifiedObjectName functionName, Optional<List<TypeSignature>> parameterTypes, AlterRoutineCharacteristics alterRoutineCharacteristics)
+    {
+        throw new PrestoException(NOT_SUPPORTED, "Alter Function is not supported in RestBasedFunctionNamespaceManager");
+    }
+
+    @Override
+    public void dropFunction(QualifiedObjectName functionName, Optional<List<TypeSignature>> parameterTypes, boolean exists)
+    {
+        throw new PrestoException(NOT_SUPPORTED, "Drop Function is not supported in RestBasedFunctionNamespaceManager");
+    }
+
+    @Override
+    public Collection<SqlInvokedFunction> listFunctions(Optional<String> likePattern, Optional<String> escape)
+    {
+        return latestFunctions.values();
```

Comment:
I think the above two methods are OK to not implement for now, but this needs to be implemented.